lib/adhearsion/calls.rb

```diff
@@ -7,15 +7,15 @@ module Adhearsion
   # This manages the list of calls the Adhearsion service receives
   class Calls
     def initialize
-      @mutex = ::Monitor.new
+      @mutex = ::Mutex.new
```

Comment:
After rebasing this on 3.0.0.rc1, all the tests pass except for this one change. I must admit that I don't really understand what the difference is. I'm also unsure if the change below is strictly related as that seems more like a pure performance tweak to avoid method_missing.

Reply:
Sorry, forgot to say how it broke. Using a Mutex first results in a weird error relating to the spec helpers but once you comment that out the block that prints the backtrack, it reveals the real error to be:

```
<ThreadError> deadlock; recursive locking
	/home/jlecuirot/Open/adhearsion/lib/adhearsion/calls.rb:80:in `synchronize'
	/home/jlecuirot/Open/adhearsion/lib/adhearsion/calls.rb:80:in `method_missing'
	/home/jlecuirot/Open/adhearsion/lib/adhearsion/calls.rb:32:in `block in remove_inactive_call'
	/home/jlecuirot/Open/adhearsion/lib/adhearsion/calls.rb:25:in `synchronize'
	/home/jlecuirot/Open/adhearsion/lib/adhearsion/calls.rb:25:in `remove_inactive_call'
	/home/jlecuirot/Open/adhearsion/lib/adhearsion/call.rb:363:in `clear_from_active_calls'
	/home/jlecuirot/Open/adhearsion/lib/adhearsion/call.rb:253:in `block in register_initial_handlers'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/has-guarded-handlers-1.6.3/lib/has_guarded_handlers.rb:162:in `call_handler'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/has-guarded-handlers-1.6.3/lib/has_guarded_handlers.rb:141:in `block (3 levels) in trigger_handler'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/has-guarded-handlers-1.6.3/lib/has_guarded_handlers.rb:136:in `catch'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/has-guarded-handlers-1.6.3/lib/has_guarded_handlers.rb:136:in `block (2 levels) in trigger_handler'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/has-guarded-handlers-1.6.3/lib/has_guarded_handlers.rb:134:in `each'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/has-guarded-handlers-1.6.3/lib/has_guarded_handlers.rb:134:in `find'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/has-guarded-handlers-1.6.3/lib/has_guarded_handlers.rb:134:in `block in trigger_handler'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/has-guarded-handlers-1.6.3/lib/has_guarded_handlers.rb:133:in `catch'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/has-guarded-handlers-1.6.3/lib/has_guarded_handlers.rb:133:in `trigger_handler'
	/home/jlecuirot/Open/adhearsion/lib/adhearsion/call.rb:201:in `block in deliver_message'
	/home/jlecuirot/Open/adhearsion/lib/adhearsion/foundation/exception_handler.rb:5:in `catching_standard_errors'
	/home/jlecuirot/Open/adhearsion/lib/adhearsion/call.rb:200:in `deliver_message'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/celluloid-0.16.0/lib/celluloid/calls.rb:26:in `public_send'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/celluloid-0.16.0/lib/celluloid/calls.rb:26:in `dispatch'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/celluloid-0.16.0/lib/celluloid/calls.rb:63:in `dispatch'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/celluloid-0.16.0/lib/celluloid/cell.rb:60:in `block in invoke'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/celluloid-0.16.0/lib/celluloid/cell.rb:71:in `block in task'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/celluloid-0.16.0/lib/celluloid/actor.rb:357:in `block in task'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/celluloid-0.16.0/lib/celluloid/tasks.rb:57:in `block in initialize'
	/home/jlecuirot/.rvm/gems/ruby-2.4.3@ahn/gems/celluloid-0.16.0/lib/celluloid/tasks/task_fiber.rb:15:in `block in create'
```

Reply:
Aha, I hadn't noticed that the rest of this file has changed quite a lot in the meantime because of de171a9669948a8bb8982c962e7c1a331fb4514b. I guess I'll just skip this commit entirely.